Create VPC with subnets and NAT gateways

JoshuaProut opened the pull request:
Creates VPC with a public and private subnet for each AZ, with IPs equally distributed. If the environment variable is Dev, only one NAT gateway is created, for all other environments a NAT gateway per subnet is created.

PAYNEA03 commented on terraform/layers/base/vpc.tf:
```diff
@@ -0,0 +1,29 @@
+
+locals {
+  azs  = data.aws_availability_zones.available.names
+  cidr = "10.0.0.0/16"
```
We are probably better capturing this in a variable. Even though the nature of VPCs mean the CIDR is unlikely to change, if we ever wanted to do a full redeploy on a different CIDR a variable would be better to change rather than directly in the code

PAYNEA03 commented on terraform/layers/base/vpc.tf:
```diff
@@ -0,0 +1,29 @@
+module "vpc" {
+  source = "terraform-aws-modules/vpc/aws"
+
+  name = "dev-vpc"
```
This would also be better as a variable, otherwise if we moved to prod the resource would be labelled dev. Also naming resources with project and environment is good practice. E.g.
name = "${var.project_name}-${var.environment}-vpc"

or creating a local, an example would be

`
locals {
  resource_name_prefix = "${var.project_name}-${var.environment}"
}
`

and naming resources like name = "${local.resource_name_prefix}-vpc"

Might help down the line - common approaches to help quickly see what reosources belong to which project / environment

JoshuaProut commented:
Moved CIDR to variables.tf to allow for better flexibility when re-creating VPC environment, added dynamic resource naming prefix to allow better identification of resources @PAYNEA03

PAYNEA03 commented on terraform/layers/base/providers.tf:
```diff
@@ -9,6 +9,8 @@ terraform {
 }
 
 provider "aws" {
+  region = "eu-west-2"
```
just spotted this, probably better as a 'region' variable. happy to be resolved on next PR
